packages/components/select/src/useSelect.ts

```diff
@@ -811,30 +809,24 @@ export const useSelect = (props, states: States, ctx) => {
       .filter((option) => option.visible)
       .every((option) => option.disabled)
   )
-
-  const navigateOptions = (direction) => {
-    if (!states.visible) {
-      states.visible = true
-      return
-    }
+  const navigateOptions = (direction: 'prev' | 'next') => {
+    states.visible = true
     if (
+      !states.visible ||
       states.options.size === 0 ||
       states.filteredOptionsCount === 0 ||
       states.isOnComposition ||
       optionsAllDisabled.value
     )
       return
 
-    if (direction === 'next') {
-      states.hoverIndex++
-      if (states.hoverIndex === states.options.size) {
-        states.hoverIndex = 0
-      }
-    } else if (direction === 'prev') {
-      states.hoverIndex--
-      if (states.hoverIndex < 0) {
-        states.hoverIndex = states.options.size - 1
-      }
+    states.hoverIndex =
```

Comment:
If the content of the nested if, I extracted to the outside.
I think the direction just has next or prev that use `? :` to replace if else